### ISSUE TITLE
dependency error when creating new nextjs project with-supertokens example

### ISSUE THREAD

**Issue by GideonStowell** [supertokens/supertokens-auth-react#565]
Running into this error when I run:
```
npx create-next-app --example with-supertokens . 
Creating a new Next.js app in /home/user/nextjs-project.

Downloading files for example with-supertokens. This might take a moment.

Installing packages. This might take a couple of minutes.

npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"^18.0.0" from the root project
npm ERR!   peer react@"^17.0.2 || ^18.0.0-0" from next@12.2.5
npm ERR!   node_modules/next
npm ERR!     next@"latest" from the root project
npm ERR!   6 more (react-dom, supertokens-auth-react, styled-jsx, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.0 || ^17.0.0" from react-select@5.2.1
npm ERR! node_modules/supertokens-auth-react/node_modules/react-select
npm ERR!   react-select@"5.2.1" from supertokens-auth-react@0.24.6
npm ERR!   node_modules/supertokens-auth-react
npm ERR!     supertokens-auth-react@"^0.24.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/user/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2022-08-20T13_16_20_115Z-debug.log

Aborting installation.
  npm install has failed.
```

`next info`:
```
Operating System:
      Platform: linux
      Arch: x64
      Version: #140~18.04.1-Ubuntu SMP Fri Aug 5 11:43:34 UTC 2022
    Binaries:
      Node: 14.15.0
      npm: 7.6.3
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 12.2.6-canary.1
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
```

If I try to force the npm install using the `--force` flag the packages will install but when I try to run the app, it will fail with this error:
```
Warning: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
TypeError: Cannot read property 'useEffect' of null
```

**Comment by GideonStowell** [comment]
Additionally:
```
> npm ls react
nextjs@ /home/gideons/options-screener/nextjs
├─┬ next@12.2.5
│ ├── react@18.2.0 deduped
│ ├─┬ styled-jsx@5.0.4
│ │ └── react@18.2.0 deduped
│ └─┬ use-sync-external-store@1.2.0
│   └── react@18.2.0 deduped
├─┬ react-dom@18.2.0
│ └── react@18.2.0 deduped
├── react@18.2.0
└─┬ supertokens-auth-react@0.24.6
  ├─┬ @emotion/react@11.10.0
  │ └── react@18.2.0 deduped
  ├─┬ react-phone-number-input@3.1.41
  │ └── react@18.2.0 deduped
  ├─┬ react-select@5.2.1
  │ ├─┬ react-dom@17.0.2
  │ │ └── react@18.2.0 deduped invalid
  │ ├─┬ react-transition-group@4.4.5
  │ │ └── react@18.2.0 deduped
  │ └── react@18.2.0 deduped invalid
  ├─┬ react-shadow@19.0.3
  │ ├─┬ react-use@15.3.8
  │ │ ├─┬ nano-css@5.3.5
  │ │ │ └── react@18.2.0 deduped
  │ │ ├─┬ react-universal-interface@0.6.2
  │ │ │ └── react@18.2.0 deduped
  │ │ └── react@18.2.0 deduped invalid
  │ └── react@18.2.0 deduped invalid
  └── react@18.2.0 deduped

npm ERR! code ELSPROBLEMS
npm ERR! invalid: react@18.2.0 /home/user/nextjs-project/node_modules/react

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2022-08-20T13_55_39_592Z-debug.log
```

**Comment by rishabhpoddar** [comment]
I think the issue is that you are using node 14. Can you try this with node 16 please?

**Comment by GideonStowell** [comment]
Upgrade to node v16.17.0 and still the same errors. I am noticing that if I start the server with `next dev` or `next` it causes this error, but running `npm run dev` it does not.

**Comment by rishabhpoddar** [comment]
Hmm. really strange. I tried:
```
yarn install
npm run dev
```

Which worked. I also tried:
```
yarn install
next dev
```

```
yarn install
next
```

```
yarn install
npx next dev
```

```
yarn install
npx next
```

All of which worked as well. i used Next.js v12.2.5. I even tried all of the above with `npm install` instead of `yarn install` and it worked.

That being said, the issue seems to be with react-select's peer dependency not matching the version of react that's being installed. Will investigate this further.
